projects/components/src/sequence/renderer/sequence-bar-renderer.service.ts

```diff
@@ -145,7 +145,7 @@ export class SequenceBarRendererService {
       )
       .attr('dy', '.82em')
       .style('font-size', '14px')
-      .text(dataRow => `${dataRow.end - dataRow.start}${options.unit !== undefined ? options.unit : ''}`);
+      .text(dataRow => `${dataRow.duration}${options.unit !== undefined ? options.unit : ''}`);
```

Comment:
Didn't get this. How is changing the duration precision need a logic change from `end - start` to `duration`?

Reply:
Both `start` and `end` only hold precision in `seconds`, @Harnoor-se7en added a new field called `durationDouble` in the Pinot table. If you will check the changes I made in other files, I am also fetching `durationDouble` from the backend. I am mapping the value of `durationDouble` to the key `duration` when the waterfall view is created.

Now, earlier, for each bar of the waterfall view, they were not explicitly using `duration`, but were rather using `endTime - startTime`, which was same as `duration`, so I made change in the `sequence-chart` view to read the value from `duration` field, which, in turn comes from `durationValue` in the backend. Let me know if you need more info around anything.